### ISSUE TITLE
gam-sae build_sae_minimal_seed: a cold hard-TopK seed routes every row to atom 0, so atoms 1..K-1 get a zero decoder and the #2822 entry gate refuses the fit

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#4519]
`crates/gam-sae/src/manifold/minimal_seed.rs`, `build_sae_minimal_seed` (main at 0b9dc2c6).

**Defect.** The cold seed builds data-driven routing only for `Softmax | OrderedBetaBernoulli`. That covers both the per-cluster periodic coordinate refinement and the `sae_residual_seed_logits` routing seed. A cold `TopK` request keeps `Array2::zeros` logits. `sae_decoder_lsq_init` evaluates the TopK gate with `topk_row`, and that function breaks ties toward the lower atom index. So every row routes to atom 0 (for `top_k = 1`), and atoms `top_k..K-1` get no rows and an identically zero LSQ decoder. The #2822 entry gate in `prepare_entry_stages` then refuses the fit:

```
atom 1 'atom_1' has an identically zero decoder. It has no column space, so its coordinates
receive no gradient and there is no data-supported decoder frame to install; seed the decoder
from the data (build_sae_minimal_seed, ...)
```

The error message recommends `build_sae_minimal_seed`, which is the very function that produced the zero decoder.

This became deterministic with acf11880 (#3090, removing the random_state logit jitter). Until then, the 1e-3 jitter split the tied rows at random between atoms. Removing the jitter was correct, but it left TopK without a real routing seed.

**Repro on main.** Both of these tests fail with the error above:
- `manifold::construction::tests_orbit_gradient_2234::two_compact_topk_orbits_are_integrated_and_differentiated_2234`
- `manifold::construction::tests_orbit_arrow_2234::the_arrow_orbit_lane_prices_two_compact_topk_orbits_2234`

Their fixture uses two periodic atoms, TopK(1), a cold seed, and two planted circles.

**Fix.** Give a cold multi-atom TopK seed the same data-driven routing Softmax and OBB get:
- The per-cluster periodic coordinate refinement.
- The mean-centred residual logits. TopK reads only their per-row order, so the atom that best explains a row wins it, and rows the data leave tied stay tied.

Claimed by wave8 session (claude/fix-topk-cold-seed-routing); draft PR incoming.
